### ISSUE TITLE
False Negative: Useless elvis is not reported by the `UnreachableCode`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6623]
## Expected Behavior
```kotlin
@Test
fun `reports unreachable code after elvis`() {
    val code = """
        fun test(): Int {
            val a = 2 ?: run {
                3
            }
        }
    """.trimIndent()
    assertThat(subject.compileAndLintWithContext(env, code)).hasSize(1)
}
```

Above TC should pass

## Observed Behavior
Above TC is failing

## Context
In my work project, we are using Detekt and have turned on the `UnreachableCode` but today I found that at some places we are using a useless elvis operator(which might get left there after some refactoring or making some variable from nullable to non null). Although this is also reported by the kotlin complier flag `allWarningsAsErrors = true` but in our project we only rely on detekt so it get slipped.

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):
